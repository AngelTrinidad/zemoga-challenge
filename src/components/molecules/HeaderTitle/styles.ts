import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  title: {
    fontWeight: '800',
  },
});

export default styles;
