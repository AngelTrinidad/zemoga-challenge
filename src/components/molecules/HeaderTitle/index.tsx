import React, {FC} from 'react';

import styles from './styles';

import {BasicHeader, BasicHeaderProps, Typography} from '~components/atoms';

interface Props extends BasicHeaderProps {
  title: string;
}

const HeaderTitle: FC<Props> = ({title, ...props}) => {
  return (
    <BasicHeader {...props}>
      <Typography variant="xxl" style={styles.title}>
        {title}
      </Typography>
    </BasicHeader>
  );
};

export default HeaderTitle;
