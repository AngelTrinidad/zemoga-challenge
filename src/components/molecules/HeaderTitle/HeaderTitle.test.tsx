import React from 'react';

import HeaderTitle from '.';

import {render} from '~helpers/testing';

describe('<HeaderTitle />', () => {
  let component: any;

  beforeEach(() => {
    component = render(<HeaderTitle title="Hello" />);
  });

  it('should render correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
