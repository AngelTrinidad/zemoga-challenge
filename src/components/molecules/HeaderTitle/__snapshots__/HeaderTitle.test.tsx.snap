// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderTitle /> should has same snapshot 1`] = `
<View
  style={
    [
      {
        "paddingHorizontal": 22.857142857142858,
        "paddingVertical": 27.42857142857143,
      },
      {
        "backgroundColor": "#CFB997",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "System",
        "fontSize": 29.485714285714288,
        "fontWeight": "800",
        "textAlign": undefined,
      }
    }
  >
    Hello
  </Text>
</View>
`;
